array_methods_lab.js

```diff
@@ -0,0 +1,308 @@
+  return currentElem % 2 === 0
+})
+console.log(itsEven)
+
+// 4. Write a function that checks if all elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+// USE EVERY ARRAY METHOD
+let sameArr = [2, 2, 2, 2, 2]
+let doubleChecked = sameArr.every((currentElem, i, arr) => {
+  if(currentElem === arr[0]){
+      return true
+    }else{
+      return false
+    }
+})
+console.log(doubleChecked)
+
+// 5. Write a function that returns the sum of every element in an array.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+// ```
+// USE THE REDUCE METHOD
+let addArr = [2,7,24,27, 5]
+let added = addArr.reduce((currentElem, i) => {
+  return currentElem + i
+})
+console.log(added)
+
+// 6. Write a function that finds largest number in an array.
+//
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+// ```
+//USE THE REDUCE METHOD
+let makeItSmall = [1, 5, 7, 8, 19, 35, 2]
+let largest = makeItSmall.reduce((accumulator,crrElem)=>{
+  if(crrElem > accumulator){
+    return crrElem
+  } else{
+    return accumulator;
+  }
+})
+console.log(largest)
+
+// 7. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+//
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+
+const countZeroes = (arr) =>{
+  return arr.reduce((acc, currentElem)=>{
+    if(currentElem === 0){ // the accumulator is going to hold all of the zeroes
+      acc++
+      // console.log("ACC: " + acc)
+    }
+    return acc
+  }, 0) // setting the initial value of the acc to zero
+}
+
+// console.log("counting ZEROS: " + countZeroes([1,0,0,2,0,3,0,0,0]))
+
+// 8. Write a function that doubles every element in an array.
+//
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+// ```
+//USE MAP METHOD
+let arr3 = [2, 4, 5, 7, 8, 9]
+let doubleUp = arr3.map((currentElem)=>{
+  return currentElem * 2
+})
+console.log(doubleUp)
+
+// 9. Write a function that finds the sum of every element in an array doubled
+//
+// ```
+// Input: [1,2,3,4]
+// Output: 20
+// ```
+let arr4 = [6, 7, 27, 24, 7, 23, 11, 4]
+let arrSum = arr4.map((secondElem)=>{
+  return secondElem * 2
+})
+.reduce((currentElem,next)=>{
+  return currentElem + next
+})
+console.log(arrSum)
+
+//simpler option
+
+const doubleSum = (arr) =>{
+  return arr.reduce((acc, currentElem)=>{
+    return acc + currentElem
+  }) * 2
+
+}
+console.log(doubleSum(arr4))
+
+// 10. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly divided by divisor.
+//
+// ```
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+// ```
+
+let arr5 = [4,5,6,7,8,9,10,11,12,13,14,15];
+const elementDivisibleBy = (divisor, arr) =>{
+  return arr.filter((currentElem)=>{
+    return currentElem % 5 === 0
+  })
+}
+console.log(elementDivisibleBy(5, arr5))
+
```

Comment:
This is a hardcoded function . If I change the divisor to 2. It still shows the 


























































































































































































































































































































































































































































































 This is a hard coded code . When I changed the divisor to 2. I still received the same answer . How can we change your function so that the divisor guides the answer for the return.